holobot/discord/data_providers/user_data_provider.py

```diff
@@ -0,0 +1,57 @@
+import hikari
+
+from holobot.discord.bot import BotAccessor
+from holobot.discord.sdk.data_providers import IUserDataProvider
+from holobot.discord.sdk.exceptions import UserNotFoundError
+from holobot.discord.sdk.models import UserData
+from holobot.discord.utils import get_guild_member_by_name
+from holobot.sdk.ioc.decorators import injectable
+from holobot.sdk.utils import assert_not_none
+
+@injectable(IUserDataProvider)
+class UserDataProvider(IUserDataProvider):
+    async def get_user_data_by_id(
```

Comment:
As discussed offline, I'll be keeping these methods as instance methods to avoid exposing implementation details to the interfaces. I'll also be keeping the invocations of static methods through the class instead of `self` to avoid giving the false impression of invoking an instance method (which may mutate the state of the instance).